### PR TITLE
Streamline encryption key and AAD generation

### PR DESCRIPTION
**Pull request by ivanyu**

### CODE REVIEW

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java [hunk 1]
```diff
@@ -69,6 +71,15 @@ public SecretKey createKey() {
         return aesKeyGenerator.generateKey();
     }
 
+    public EncryptionKeyAndAAD createKeyAndAAD() {
```
Comment (jeqo) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java:74]: Should we rename the returned class to `SecretKeyAndAAD` instead to keep it aligned with `createKey(): SecretKey`?

**Review by jeqo — COMMENTED** [review]
LGMT, adding a small comment about naming

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyAndAAD.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final EncryptionKeyAndAAD that = (EncryptionKeyAndAAD) o;
+        return Objects.equals(key, that.key) && Arrays.equals(aad, that.aad);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(key);
```
Comment (AnatolyPopov) [inline commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyAndAAD.java:47]: I guess this instead
```suggestion
        int result = Objects.hashCode(key);
```

File: commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java (hunk 1 above)
Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java:74]: > Should we have a test for it also?

Not sure. What to test, that it doesn't return `null`?

Comment (ivanyu) [reply commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java:74]: > Should we rename the returned class to `SecretKeyAndAAD` instead to keep it aligned with `createKey(): SecretKey`?

Unification of these two terms throughout the code base is indeed needed. I personally would prefer "encryption" over "secret". But this probably goes beyond the scope of this PR. I've made https://github.com/aiven/tiered-storage-for-apache-kafka/issues/166 for this.

**Pull request merged**